examples/spotify_example.py

```diff
@@ -6,46 +6,147 @@
+if cast.cast_type=='group':
+    # The spotify API will return the name of the group controller device, not 
+    # the group we want to cast to. Setup a MultiZone controller to map devices
+    # returned by the Spotify API to members of the group we want to cast to.
+    mz = MultizoneController(cast.uuid)
+    mz_listener = MzListener()
+    mz.register_listener(mz_listener)
+    cast.register_handler(mz)
 
-    client = spotipy.Spotify(auth=access_token)
+    # Register a connection listener, when connected, we can poll group members
+    cast.register_connection_listener(ConnListener(mz))
+    
+# Wait for connection to the chromecast
+cast.wait()
 
-    sp = SpotifyController(access_token, expires)
-    cast.register_handler(sp)
-    sp.launch_app()
+device_id = None
 
-    devices_available = client.devices()
+# Create a spotify token
+data = st.start_session(args.user, args.password)
+access_token = data[0]
+expires = data[1] - int(time.time())
 
+# Create a spotify client
+client = spotipy.Spotify(auth=access_token)
+if args.show_debug:
+    spotipy.trace = True
+    spotipy.trace_out = True
+
+# Launch the spotify app on the cast we want to cast to
+sp = SpotifyController(access_token, expires)
+cast.register_handler(sp)
+sp.launch_app()
+
+# Query spotify for active devices
+devices_available = client.devices()
+
+# If we're not casting to a group
+if mz == None:
+    # Try to match active spotify devices with the cast we want to cast to
+    # Ideally we would match device['id'] with the cast's UUID because there
+    # may be multiple casts with the same name. Currently it's not known how to
+    # map between the two so match name instead.
     for device in devices_available['devices']:
-        if device['name'] == CAST_NAME:
+        if device['name'] == args.cast:
             device_id = device['id']
             break
 
-    client.start_playback(device_id=device_id, uris=["spotify:track:3Zwu2K0Qa5sT6teCCHPShP"])
+    if not device_id:
+        print('No device with name "{}" known by Spotify'.format(args.cast))
+        print('Known devices: {}'.format(devices_available['devices']))
+        sys.exit(1)
+else:
+    # Wait for group members to update
+    while not mz_listener.got_members:
+        time.sleep(0.1)
+
+    # Look for a chromecast which:
+    # - Is known to spotify (matched by bame)
+    # - Is a member of the audio group we want to play to
+    for device in devices_available['devices']:
+        for _cast in chromecasts:
+            if str(_cast.uuid) in mz.members and device['name'] == _cast.name:
+                device_id = device['id']
+                break
+
+    if not device_id:
+      print('Could not find matching member of audio group')
+      print('Known devices: {}'.format(devices_available['devices']))
+      print('Members of group "{}": {}'.format(args.cast, mz.members))
+      sys.exit(1)
+
+# Start playback
+client.start_playback(device_id=device_id, uris=[args.track])
```

Comment:
It would be nice to support playlists as well like
```
if args.uri.find('track') > 0:
    client.start_playback(device_id=spotify_device_id, uris=[args.uri])
else:
    client.start_playback(device_id=spotify_device_id, context_uri=args.uri)
```
And change args accordingly:
```
parser.add_argument('--uri', help='Spotify uri (default: "%(default)s")',
                    default="spotify:track:3Zwu2K0Qa5sT6teCCHPShP")
```